src/Trigger.cc

```diff
@@ -218,7 +218,21 @@ bool Trigger::Eval()
 	// An alternative approach to copying the frame would be to deep-copy
 	// the expression itself, replacing all references to locals with
 	// constants.
-	Frame* f = frame->Clone();
+
+	Frame* f = nullptr;
+
+	try
+		{
+		f = frame->Clone();
+		}
+	catch ( InterpreterException& )
+		{
+		// Frame contains values that couldn't be cloned. It's
+		// already been reported, disable trigger.
+		Disable();
+		return false;
```

Comment:
I'm going to add an `Unref(this)` after the `Disable()`.  Other usages do the same and I can trace out at least one code path that otherwise leaks the `Trigger` from here: walk through what ref-count is after the `Trigger()` ctor and after the timer expires (since there's no other global structure holding a reference at that time).